pkg/external-provider/external_series_registry.go

```diff
@@ -0,0 +1,109 @@
+package provider
+
+import (
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/labels"
+
+	prom "github.com/directxman12/k8s-prometheus-adapter/pkg/client"
+)
+
+// ExternalSeriesRegistry acts as the top-level converter for transforming Kubernetes requests
+// for external metrics into Prometheus queries.
+type ExternalSeriesRegistry interface {
+	// ListAllMetrics lists all metrics known to this registry
+	ListAllMetrics() []provider.ExternalMetricInfo
+	QueryForMetric(namespace string, metricName string, metricSelector labels.Selector) (prom.Selector, bool, error)
+}
+
+// overridableSeriesRegistry is a basic SeriesRegistry
+type externalSeriesRegistry struct {
+	mu sync.RWMutex
```

Comment:
could you document what fields are protected by this?

Reply:
Preferred way to document this? Inline comment ?

Reply:
yes, a small comment is totall sufficient :+1:

Reply:
@antoinne85 or @DirectXMan12 
From what I can tell we lock the metrics series when we read to prevent read / write issues. Can you just confirm so I don't mislead :)